datavault-common/src/main/java/org/datavaultplatform/common/model/dao/PermissionDAOImpl.java

```diff
@@ -0,0 +1,75 @@
+package org.datavaultplatform.common.model.dao;
+
+import org.datavaultplatform.common.model.Permission;
+import org.datavaultplatform.common.model.PermissionModel;
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.criterion.Restrictions;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class PermissionDAOImpl implements PermissionDAO {
+
+    private SessionFactory sessionFactory;
+
+    public PermissionDAOImpl() {}
+
+    public void setSessionFactory(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void synchronisePermissions() {
+        Session session = this.sessionFactory.openSession();
+        Transaction tx = session.beginTransaction();
+
+        // First, remove any permissions that are not found in the Permission enum...
+        String commaSeparatedPermissionIds = Arrays.stream(Permission.values())
```

Comment:
Can we pass in an array/use the JPA criteria api to do this? Generating SQL strings from string concatenation is going to flag up in any analysis tool and means we can't statically verify the SQL.

Reply:
I struggled to get that working last week (which is why it's using string concatenation) but I'll have another look at doing this with a `Criteria`

Reply:
Let's not spend too much time on it. At the moment, it's just a code smell. The likelihood of someone exploiting the database be changing the bytecode of an enum class to contain a SQL injection string is... low.